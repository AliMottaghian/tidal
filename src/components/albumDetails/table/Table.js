import PropTypes from "prop-types";
import { timeFormat } from "../../../utils/index";
import classes from "./table.module.css";

const Table = ({ tracks, releaseDate }) => {
  const renderTrackList = () => {
    return tracks.map((track, index) => {
      return (
        <tr key={index}>
          <td></td>
          <td>{index + 1}</td>
          <td>{track.title}</td>
          <td>{track.artist.name}</td>
          <td>{timeFormat(+track.duration)}</td>
          <td>{releaseDate}</td>
        </tr>
      );
    });
  };

  return (
    <table className={classes.table}>
      <thead>
        <tr>
          <th className={classes.emptyHeader}></th>
          <th>#</th>
          <th width="30%">Title</th>
          <th>Artist</th>
          <th>Time</th>
          <th>Released</th>
        </tr>
      </thead>
      <tbody>{renderTrackList()}</tbody>
    </table>
  );
};

Table.propTypes = {
  tracks: PropTypes.array,
  releaseDate: PropTypes.string,
};
export default Table;
