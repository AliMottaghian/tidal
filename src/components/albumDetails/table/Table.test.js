import Table from "./Table";

describe("<Table />", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
      <Table
        tracks={[
          { title: "test1", artist: { name: "artist name1" }, duration: 300 },
          { title: "test2", artist: { name: "artist name2" }, duration: 500 },
        ]}
      />
    );
  });
  it("should render Table component without crashing", () => {
    expect(wrapper).toMatchSnapshot();
  });
  it("should render render three <tr>", () => {
    const tableRow = wrapper.find("tr");
    expect(tableRow).toHaveLength(3);
  });
});
