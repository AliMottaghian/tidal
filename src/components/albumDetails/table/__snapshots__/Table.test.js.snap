// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Table /> should render Table component without crashing 1`] = `
<table
  className="table"
>
  <thead>
    <tr>
      <th
        className="emptyHeader"
      />
      <th>
        #
      </th>
      <th
        width="30%"
      >
        Title
      </th>
      <th>
        Artist
      </th>
      <th>
        Time
      </th>
      <th>
        Released
      </th>
    </tr>
  </thead>
  <tbody>
    <tr
      key="0"
    >
      <td />
      <td>
        1
      </td>
      <td>
        test1
      </td>
      <td>
        artist name1
      </td>
      <td>
        05:00
      </td>
      <td />
    </tr>
    <tr
      key="1"
    >
      <td />
      <td>
        2
      </td>
      <td>
        test2
      </td>
      <td>
        artist name2
      </td>
      <td>
        08:20
      </td>
      <td />
    </tr>
  </tbody>
</table>
`;
